### DISCUSSION TITLE
What does "dot repeatable jumps" mean?

### DISCUSSION THREAD

**Discussion by GreenSleeper** [folke/flash.nvim#65] [Q&A]
What does "dot repeatable jumps" mean?
What are the use cases?

**Comment by folke** [comment]
`ds` starts flash, you pick a label and everyting up to the target gets deleted.
`.` will open flash again with the same pattern so you can pick a new label

for `ftFT` it will autmoatically repeat the jump

**Comment by GreenSleeper** [comment]
I'm using `LazyVim` with the default configuration for `flash.nvim`.

> `ds` starts flash, you pick a label and everyting up to the target gets deleted.
`.` will open flash again with the same pattern so you can pick a new label

when I hit `.`, it won't let me pick a new label, it just keep deleting.

> for `ftFT` it will autmoatically repeat the jump

`.` does just what `.` usually does rather than repeating the jump.

**Reply by folke** [reply]
It sounds like you're not using the lazyvim flash extra.
If you configured flash yourself, make sure to disable leap and flit.

**Reply by GreenSleeper** [reply]
I'm using the lazyvim falsh extra.
leap and flit are disabled automaticlly.

**Comment by GreenSleeper** [comment]
And if I add the configuration below, `;`, `,` stop working for `fFtT`
```lua
{
    "folke/flash.nvim",
    init = function ()
      -- to use this, make sure to set `opts.modes.char.enabled = false`
      local Config = require("flash.config")
      local Char = require("flash.plugins.char")
      for _, motion in ipairs({ "f", "t", "F", "T" }) do
        vim.keymap.set({ "n", "x", "o" }, motion, function()
          require("flash").jump(Config.get({
            mode = "char",
            search = {
              mode = Char.mode(motion),
              max_length = 1,
            },
          }, Char.motions[motion]))
        end)
      end
    end,

    opts = {
      modes = {
        char = {
          enabled = false,
        },
      },
    },
  }
```

**Reply by folke** [reply]
Yes, obviously, since you're not using the default config and didnt map `;` and `,`. There's also no code to make it repeatable here

**Reply by folke** [reply]
I'll see to make jump targets configurable for ftFT

**Comment by folke** [comment]
I just pushed a lot of new changes and smart labeled ftFT searches are now included.

You can repeat a jump the same way as without labels using `f` and `t` for next, `F` and `T` for prev. 

Use the below configuration.

```lua
 opts = {
      modes = {
        char = {
          jump_labels = function(motion)
            -- never show jump labels by default
            -- return false
            -- Always show jump labels for ftFT
            return vim.v.count == 0 and motion:find("[ftFT]")
            -- Show jump labels for ftFT in operator-pending mode
            -- return vim.v.count == 0 and motion:find("[ftFT]") and vim.fn.mode(true):find("o")
          end,
        },
      },
    },
```

It combines ftFT searches with labels. You can still navigate clever-f style or use `;`, `,` as well. A lot of labels are also excluded, so you can continue from the new position with any key like `hjklia...`

It pretty much combines the best of both worlds :)
you can also only enable it for operator pending mode or any other conditions.

Let me know what you think!

**Comment by GreenSleeper** [comment]
Yes, now `;` `,` works with labels and this works for me too:
> ds starts flash, you pick a label and everyting up to the target gets deleted.
> . will open flash again with the same pattern so you can pick a new label

wonderful, thank you for such great work!
